lib/code_corps/validators/slug_validator.ex

```diff
@@ -0,0 +1,31 @@
+defmodule CodeCorps.Validators.SlugValidator do
+  @moduledoc """
+  Used for validating slug fields in a given changeset.
+  """
+
+  alias Ecto.Changeset
+
+  def validate_slug(changeset, field_name) do
+    valid_slug_pattern = ~r/\A((?:(?:(?:[^-\W]-?))*)(?:(?:(?:[^-\W]-?))*)\w+)\z/
```

Comment:
Maybe not. Probably just personal preference :) I don't like dealing with regex so I try adding comments when I can so you don't have to go looking for tests. Kinda make it part of the documentation for the method.